src/main/java/org/fiware/iam/tmforum/handlers/QuoteEventHandler.java

```diff
@@ -0,0 +1,196 @@
+package org.fiware.iam.tmforum.handlers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.micronaut.http.HttpResponse;
+import jakarta.inject.Singleton;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.fiware.iam.PriceMapper;
+import org.fiware.iam.configuration.GeneralProperties;
+import org.fiware.iam.domain.Policy;
+import org.fiware.iam.dsp.RainbowAdapter;
+import org.fiware.iam.exception.RainbowException;
+import org.fiware.iam.tmforum.TMForumAdapter;
+import org.fiware.iam.tmforum.quote.model.*;
+import org.fiware.rainbow.model.NegotiationRequestVO;
+import org.fiware.rainbow.model.ObligationVO;
+import org.fiware.rainbow.model.OfferVO;
+import org.fiware.rainbow.model.PermissionVO;
+import reactor.core.publisher.Mono;
+
+import java.util.*;
+import java.util.stream.Stream;
+
+import static org.fiware.iam.PriceMapper.PAYMENT_ACTION;
+import static org.fiware.iam.tmforum.TMForumAdapter.CONSUMER_ROLE;
+
+/**
+ * Handle all incoming events in connection to Quote
+ */
+@RequiredArgsConstructor
+@Singleton
+@Slf4j
+public class QuoteEventHandler implements EventHandler {
+
+	private static final String URN_UUID_TEMPLATE = "urn:uuid:%s";
+
+
+	private static final String CREATE_EVENT = "QuoteCreateEvent";
+	private static final String DELETE_EVENT = "QuoteDeleteEvent";
+	private static final String STATE_CHANGE_EVENT = "QuoteStateChangeEvent";
+	private static final String ATTRIBUTE_CHANGE_EVENT = "QuoteAttributeValueChangeEvent";
+
+	private static final List<String> SUPPORTED_EVENT_TYPES = List.of(CREATE_EVENT, DELETE_EVENT, STATE_CHANGE_EVENT, ATTRIBUTE_CHANGE_EVENT);
+
+	private final ObjectMapper objectMapper;
+	private final GeneralProperties generalProperties;
+	private final PriceMapper priceMapper;
+	private final TMForumAdapter tmForumAdapter;
+	private final RainbowAdapter rainbowAdapter;
+
+
+	@Override
+	public boolean isEventTypeSupported(String eventType) {
+		return SUPPORTED_EVENT_TYPES.contains(eventType);
+	}
+
+	@Override
+	public Mono<HttpResponse<?>> handleEvent(String eventType, Map<String, Object> event) {
+		return switch (eventType) {
+			case CREATE_EVENT -> handleQuoteCreation(event);
+			case STATE_CHANGE_EVENT, ATTRIBUTE_CHANGE_EVENT -> handleQuoteStateChange(event);
+			case DELETE_EVENT -> handleQuoteDeletion(event);
+			default -> throw new IllegalArgumentException("Even type %s is not supported.".formatted(eventType));
+		};
+	}
+
+	private Mono<HttpResponse<?>> handleQuoteCreation(Map<String, Object> event) {
+		QuoteCreateEventVO quoteCreateEventVO = objectMapper.convertValue(event, QuoteCreateEventVO.class);
+		QuoteVO quoteVO = quoteCreateEventVO.getEvent()
+				.getQuote();
+		List<PermissionVO> permissionVOS = new ArrayList<>(getPermissionsFromQuote(quoteVO));
+
+		return tmForumAdapter.getConsumerDid(quoteVO)
+				.flatMap(id -> rainbowAdapter.createParticipant(id, CONSUMER_ROLE))
```

Comment:
Similar question as with the producer participant

Reply:
Same answer:)